design-proposals/sdn-ipam-secondary-nets.md

```diff
@@ -0,0 +1,450 @@
+# Overview
+
+## Motivation
+Virtual Machine owners want to offload IPAM from their custom solutions (e.g.
+custom DHCP server running on their cluster network) to SDN.
+
+The other alternative - static IP address allocation - is also not ideal for
+virtualization users, given how cumbersome it is - i.e. specifying the IP for
+each VM's interface.
+
+## Definition of users
+There are two types of users regarding this feature, the network admin, and the
+VM user. They are described below:
+- net admin: this user can configure the network interconnecting the Kubernetes
+nodes. They can create / update / delete network attachment definitions.
+- VM user: this user can create / delete virtual machines.
+
+## User Stories
+- As a VM user, I want my VMs to automatically get IP addresses from a
+configured range, so I don't need to manage my own IP pools, DHCP server, or
+static IP configuration.
+- As a VM user, I want to migrate (one of) my VMs to another node, preserving
+the networking configuration, so the IP-aware applications running in the VM are
+not impacted.
+- As a VM user, I want to start Virtual Machines (currently stopped) with the
+original VM network configuration, so the IP-aware applications running in the
+VM are not impacted.
+- As a network admin, I want to decide which networks can be subject to having
+persistent IP allocations surviving VM migration, and VM stop.
+
+## Goals
+- Manage KubeVirt Virtual Machines IP addresses from within the SDN solution
```

Comment:
Hm, I actually am saying
> Manage KubeVirt Virtual Machines IP addresses from within the SDN solution

thus, KubeVirt does not manage IP addresses. The SDN manages addresses for kubevirt vms IP addresses. 
... or for whoever asks for it. But since this is from a Kubevirt's perspective ...